lib/src/main/java/com/dynatrace/file/util/FilePollerFactory.java

```diff
@@ -0,0 +1,64 @@
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FilePollerFactory {
+  private static final Logger logger = Logger.getLogger(FilePollerFactory.class.getName());
+  private static final boolean isMacOs =
+      System.getProperty("os.name").toLowerCase().contains("mac");
```

Comment:
Can this ever be not set or null?
You could check out the impl of https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/SystemUtils.html#IS_OS_MAC for this.

Reply:
I added a default so it never returns null, but returns an empty string instead. This way, if the os.name property is not set there is no NullpointerException, but it simply returns false (which would lead to the default behavior of using the WatchService based FilePoller)